patches/chrome-tools-build-win-create_installer_archive.py.patch

```diff
@@ -1,33 +1,50 @@
 diff --git a/chrome/tools/build/win/create_installer_archive.py b/chrome/tools/build/win/create_installer_archive.py
-index ffbf9ec0727b9a81eb4b8488ae67b797ec81a437..cdd0f6769f6629b855a56f267724673a79fec836 100755
+index ffbf9ec0727b9a81eb4b8488ae67b797ec81a437..444a2a1bf4776f7db7f209d8a7b7ab8d55b54bb7 100755
 --- a/chrome/tools/build/win/create_installer_archive.py
 +++ b/chrome/tools/build/win/create_installer_archive.py
-@@ -20,7 +20,7 @@ import os
- import shutil
- import subprocess
- import sys
--
+@@ -42,6 +42,7 @@ SETUP_PATCH_FILE_PREFIX = "setup_patch"
+ TEMP_ARCHIVE_DIR = "temp_installer_archive"
+ VERSION_FILE = "VERSION"
+ 
 +from create_installer_archive_helper import SignAndCopyPreSignedBinaries, BraveCopyAllFilesToStagingDir
  
- ARCHIVE_DIR = "installer_archive"
+ g_archive_inputs = []
  
-@@ -112,6 +112,7 @@ def CopyAllFilesToStagingDir(config, distribution, staging_dir, build_dir,
+@@ -112,6 +113,7 @@ def CopyAllFilesToStagingDir(config, distribution, staging_dir, build_dir,
    if enable_hidpi == '1':
      CopySectionFilesToStagingDir(config, 'HIDPI', staging_dir, build_dir,
                                   verbose)
 +  BraveCopyAllFilesToStagingDir(config, staging_dir, g_archive_inputs)
  
    if include_snapshotblob == '1':
      CopySectionFilesToStagingDir(config, 'SNAPSHOTBLOB', staging_dir, build_dir,
-@@ -551,6 +552,7 @@ def main(options):
+@@ -155,7 +157,7 @@ def CopySectionFilesToStagingDir(config, section, staging_dir, src_dir,
+ 
+ def GenerateDiffPatch(options, orig_file, new_file, patch_file):
+   if (options.diff_algorithm == "COURGETTE"):
+-    exe_file = os.path.join(options.last_chrome_installer, COURGETTE_EXEC)
++    exe_file = os.path.join(options.build_dir, COURGETTE_EXEC)
+     cmd = '%s -gen "%s" "%s" "%s"' % (exe_file, orig_file, new_file, patch_file)
+   else:
+     exe_file = os.path.join(options.build_dir, BSDIFF_EXEC)
+@@ -179,7 +181,7 @@ def GetPrevVersion(build_dir, temp_dir, last_chrome_installer, output_name):
+                                    output_name + ARCHIVE_SUFFIX)
+   cmd = [lzma_exec,
+          'x',
+-         '-o"%s"' % temp_dir,
++         '-o%s' % temp_dir,
```

Comment:
W/o this change, `lzma_exec` generates temp_dir under `c:/`.

Reply:
Is it ever possible for `temp_dir` to have spaces? If so, you might try something like `-o.\"' % temp_dir`

Reply:
Yeah, thanks for catching. It would be much safer.
Changed to `-o./"%s" % temp_dir`.

Reply:
I still don't understand why this is needed, how does it work for chrome?

Reply:
After discussing with @bridiver , This is reverted to `'-o%s' % temp_dir,` because relative path in out doesn't have space.